doc/topics/releases/2015.8.4.rst

```diff
@@ -0,0 +1,15 @@
+===========================
+Salt 2015.8.4 Release Notes
+===========================
+
+Core Changes
+============
+
+- Added ``state.validate`` feature.  This checks SLS data against the salt
+  state API and reports errors if the argument types do not match the types
```

Comment:
It checks the named args only and the types of their default values if those are provided in the function signature.  I know of no way to process kwargs.  If you have an idea for how to do this, then I can work on adding that.